### ISSUE TITLE
Adding one icon adds all shapes from set

### ISSUE THREAD

**Issue by mme-private** [vmware/clarity#2599]
## Describe the bug

Adding one clarity icon increases main.js bundle size (ng build --prod) by ~260KB

## How to reproduce

1. Create new project and add new dependencies
```bash
ng new my-app --style=scss
cd my-app
yarn add @clr/angular @clr/icons @clr/ui bootstrap@4.0.0-alpha.5 @webcomponents/custom-elements@1.0.0
```

2. Change angular.json to
```json
"styles": [
  "node_modules/@clr/icons/clr-icons.min.css",
  "src/styles.scss"
],
"stylePreprocessorOptions": {
  "includePaths": [
  "node_modules"
  ]
},
  "scripts": [
   "node_modules/@webcomponents/custom-elements/custom-elements.min.js"
  ]
```
3. Change style.css to
```scss
@import "@clr/ui/src/utils/dependencies.clarity";
@import "@clr/ui/src/utils/components.clarity";
```
4. Change app-module.ts to
```typescript
import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';

import { ClarityModule } from '@clr/angular';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { AppComponent } from './app.component';

@NgModule({
  declarations: [
    AppComponent
  ],
  imports: [
    BrowserModule,
    ClarityModule,
    BrowserAnimationsModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
```

5. Run build for the first time
```bash
ng build --prod
```
6. **Check ./dist/my-app/main.js size (main.d75896a6224663291cac.js (main) 308 kB)**
7. Change app.component.ts to
```typescript
import { Component } from '@angular/core';

import { ClarityIcons } from '@clr/icons';
import { ClrShapeTrash } from '@clr/icons/shapes/essential-shapes';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss']
})
export class AppComponent {
  title = 'my-app';

  constructor() {

    ClarityIcons.add({
      trash: ClrShapeTrash,
    });
  }
}
```
8. Run build second time
```bash
ng build --prod
```
9. **Check ./dist/my-app/main.js size (main.f524c5eda53ee31939a4.js (main) 574 kB)** 

So size increment is more than 260KB.
After checking content of main.js, there is visible that large part of file consists of SVG shapes (... t.ClrShapeCog=l.clrIconSVG( ... ), so after adding one icon, final file contains all set icon paths.

## Expected behavior

Ideally main.js should contain only one icon path.

## Versions

**App**

* Angular: 6.1.4
* Node: 8.9.0
* Clarity: 0.12.10

**Device:**

* Type: PC
* OS: Windows 7 Ultimate SP1
* Browser Chrome
* Version 68

**Comment by gnomeontherun** [comment]
Currently adding any shape from a shape set is automatically adding all of the shapes from that set. See https://github.com/vmware/clarity/blob/master/src/clr-icons/shapes/essential-shapes.ts#L984. We need to rethink and fix this but is probably a breaking change so not a simple fix at the moment.

**Comment by Shijir** [comment]
@metimus thanks for filing this issue in detail. I confirm this is a bug. As @gnomeontherun mentioned this solution would likely require a breaking change.

**Comment by gnomeontherun** [comment]
This will be resolved by #2821 and part of 1.0.

**Comment by gnomeontherun** [comment]
Reopening as part of 2.0 due to blocking issues to resolve.

**Comment by gnomeontherun** [comment]
It was also noted that if you include a specific shape set once and also the all-shapes set elsewhere in the same application (or if a dependency includes a set), the result will not tree shake the individual shape sets in favor of all-shapes set. Advice is to not use all shape sets unless you import all shapes once and never import other sets individually elsewhere.

**Comment by adumesny** [comment]
trying to do some tree shaking of clarity icons (and eventually in an exported lib), but for now I tried a clean cli 7project, added clarity (`ng add @clr/angular`), *removed node_modules/@clr/icons/clr-icons.min.js* and instead added this code in app.component.ts

```
import { ClarityIcons } from '@clr/icons';
import { ClrShapePencil } from '@clr/icons/shapes/essential-shapes';
ClarityIcons.add({
  'pencil': ClrShapePencil,
});
```
and I can tell it's needed as pencil icon is missing otherwise.
build using `ng build --prod --stats-json` but webpack-bundle-analyzer still tells me it's bring entire 165k essential-shapes !

**Comment by mathisscott** [comment]
Fixed with #3967

**Comment by mattmutt** [comment]
Can the latest fix #3967 be backported from Clarity core, or is there a way we can explicitly call out which SVG definitions are used?

The goal it to mimic font subsetting in a similar approach but with icons. The excessive SVG markup is indeed contributing to code bloat, if I hand pick just a few icons from the essential shapes.

**Comment by mathisscott** [comment]
@mattmutt 
No, it can't be backported. If you are only using a few icons from the essential shapes though, you could theoretically move over to `@clr/core` icons once those shapes have been implemented there.

`@clr/core` has no dependency on angular or clarity versions. so you could use it in isolation in place of `@clr/icons`.
